source/opencascade/boundary_lib.cc

```diff
@@ -233,14 +233,11 @@ namespace OpenCASCADE
   ArclengthProjectionLineManifold<dim,spacedim>::pull_back(const Point<spacedim> &space_point) const
   {
     double t (0.0);
-#ifdef DEBUG
     ShapeAnalysis_Curve curve_analysis;
     gp_Pnt proj;
-    Assert(curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true)
-           < tolerance*length, ExcPointNotOnManifold(space_point));
-#else
-    (void)space_point;
-#endif
+    double dist = curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true);
+    Assert(dist < tolerance*length, ExcPointNotOnManifold(space_point));
+    dist *= 2; // Silence compiler warning in Release mode.
```

Comment:
Just change this to `(void)dist;` and make the variable `const` -- this is the standard idiom we use everywhere. Then please merge!